std/traits.d

```diff
@@ -5998,6 +6000,60 @@ enum bool isNarrowString(T) = (is(T : const char[]) || is(T : const wchar[])) &&
     }
 }
 
+/**
+ * Detects whether `T` is a comparable type. Basic types and structs and
+ * classes that implement opCmp are ordering comparable. All user defined
+ * types are equality comparable.
+ */
+enum bool isOrderingComparable(T) = is(typeof((T t) => t < t));
```

Comment:
Needs to test whether `t < t` can be tested with `if`. Use https://dlang.org/library/std/traits/if_testable.html